controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -373,52 +342,77 @@ public static long getSizeTillStreamCut(final byte[] indexTable, final byte[] hi
         }
     }
 
+    /**
+     * Add new segments to the segment table.
+     * This method is designed to work with chunked creation. So it takes a
+     * toCreate count and newRanges and it picks toCreate entries from the end of newranges.
+     *
+     * @param newRanges             ranges
+     * @param timeStamp             timestamp
+     * @return serialized segment table
+     */
+    public static byte[] createSegmentTable(final List<AbstractMap.SimpleEntry<Double, Double>> newRanges, final long timeStamp) {
+        final ByteArrayOutputStream segmentStream = new ByteArrayOutputStream();
+        try {
+            writeSegmentsToSegmentTable(0, 0, newRanges, timeStamp, segmentStream);
+        } catch (Exception e) {
```

Comment:
If writeSegmentsToSegmentTable was annotated, then the catch would not be needed here.

Reply:
done